src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/JsonReaderDelegator.cs

```diff
@@ -153,37 +153,29 @@ internal static DateTime ParseJsonDateInDefaultFormat(string originalDateTimeVal
                 throw new FormatException(SR.Format(SR.JsonInvalidDateTimeString, originalDateTimeValue, JsonGlobals.DateTimeStartGuardWriter, JsonGlobals.DateTimeEndGuardWriter));
             }
 
-            string ticksvalue = dateTimeValue.Substring(6, dateTimeValue.Length - 8);
+            ReadOnlySpan<char> ticksvalue = dateTimeValue.AsSpan(6, dateTimeValue.Length - 8);
             long millisecondsSinceUnixEpoch;
             DateTimeKind dateTimeKind = DateTimeKind.Utc;
-            int indexOfTimeZoneOffset = ticksvalue.IndexOf('+', 1);
+            int indexOfTimeZoneOffset = ticksvalue.Slice(1).IndexOf('+');
 
             if (indexOfTimeZoneOffset == -1)
             {
-                indexOfTimeZoneOffset = ticksvalue.IndexOf('-', 1);
+                indexOfTimeZoneOffset = ticksvalue.Slice(1).IndexOf('-');
             }
 
-            if (indexOfTimeZoneOffset != -1)
+            if (indexOfTimeZoneOffset >= 0)
             {
                 dateTimeKind = DateTimeKind.Local;
-                ticksvalue = ticksvalue.Substring(0, indexOfTimeZoneOffset);
+                ticksvalue = ticksvalue.Slice(0, 1 + indexOfTimeZoneOffset);
```

Comment:
Subtle. Feel like this needs a comment as to why the `1 + ...` is there. (Presumably it's to counter the `Slice(1)` calls above.)